go/fn/object.go

```diff
@@ -981,6 +988,11 @@ func (o *SubObject) GetSlice(k string) SliceSubObjects {
 	return o.NestedSliceOrDie(k)
 }
 
+// SetSlice set a slice of SubObject to the given fieldPath. It upsert the existing slice value.
+func (o *SubObject) SetSlice(objects SliceSubObjects, fields ...string) error {
+	return o.SetNestedField(objects, fields...)
```

Comment:
`SetSlice` is specifically for `SubObject`, like `GetMap` `GetSlice` functions，so I think calling `SetNestedField` here can over complicate the workflow.  For example, the `SetNestedField` has `onLockedFields` checks and runs different workflows for different interface type. All these are not necessary for `SetSlice` function.